lit_gpt/lora.py

```diff
@@ -136,8 +137,23 @@ def reset_parameters(self) -> None:
     def merge(self) -> None:
         """Merges the LoRA weights into the full-rank weights (W = W + delta_W)."""
         if self.r > 0 and not self.merged:
-            # Merge the weights and mark it
-            self.linear.weight.data += (self.lora_B @ self.lora_A) * self.scaling
+            lora_data = (self.lora_B @ self.lora_A) * self.scaling
+            # if the pretrained weights are not quantized - simply sum them with LoRA weights
+            if self.linear.weight.data.dtype == lora_data.dtype:
```

Comment:
AFAIK, the dtype might be different only when the pretrained weights are quantized.
But maybe I'm missing something?

Looks clean enough + supports cases when the dtypes are equal (like in QA-LoRA when the pretrained and LoRA weights are quantized).

Reply:
Decided to make it more explicit and made quantization happening only when the dtype of the pretrained weights is `torch.uint8` and dtype of LoRA weights – isn't.